### ISSUE TITLE
UnicodeDecodeError in Python 3.5

### ISSUE THREAD

**Issue by lahwaacz** [earwig/mwparserfromhell#125]
I'm getting this exception in Python 3.5 when parsing [this page](https://wiki.archlinux.org/index.php?title=AMule_(%E7%AE%80%E4%BD%93%E4%B8%AD%E6%96%87)&oldid=401407):
```
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-4: ordinal not in range(128)

During handling of the above exception, another exception occurred:

SystemError: <built-in method lower of str object at 0x7f7e43c4b030> returned a result with an error set

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 11, in <module>
    mwparserfromhell.parse(text)
  File "/usr/lib/python3.5/site-packages/mwparserfromhell/utils.py", line 58, in parse_anything
    return Parser().parse(value, context, skip_style_tags)
  File "/usr/lib/python3.5/site-packages/mwparserfromhell/parser/__init__.py", line 93, in parse
    tokens = self._tokenizer.tokenize(text, context, skip_style_tags)
SystemError: <class 'mwparserfromhell.parser.tokens.TagOpenOpen'> returned a result with an error set
```
How can I help debugging this? It's the C tokenizer again and I have no idea where to start...

**Comment by earwig** [comment]
Confirming this is Python 3 + ctokenizer-specific. Works on Python 2 or using the pytokenizer.

**Comment by lahwaacz** [comment]
It seems that the original exception happens in the [unicode_to_lcase_ascii](https://github.com/earwig/mwparserfromhell/blob/develop/mwparserfromhell/parser/ctokenizer/definitions.c#L62) function in `definitions.c`. Any idea how to fix it?

**Comment by earwig** [comment]
I'm working on it.

**Comment by earwig** [comment]
Fixed in 23d9758. Turns out it was a regression from #97. Thanks for your help.
